docs/design/fabric-v2+/fpc-key-dist.puml

```diff
@@ -213,111 +181,135 @@ group key distribution
     true/false (say h==H(value of store key)).
   end note
   return true /'Enclave1 -> TLCC_Enclave1'/
-    note left
-        if false, abort
-    end note
-  Enclave1      -> TLCC_Enclave1    ++: can_endorse(ECC2_S_PK)
-    note right
-        TLCC checks the endorsement policy to determine
-        if the recipient enclave is an eligible endorser.
-    end note
-    note right
-        TLCC should check that the recipient enclave's MRENCLAVE
-        matches the chaincode definition's MRENCLAVE.
-        This could enable key distribution from an old cc version,
-        with a different MRENCLAVE.
-        Notice: in this case there cannot be also a check
-        can_endorse(ECC1_S_PK) -- or it should be negated.
-    end note
+  note left
+    if false, abort
+  end note
+  Enclave1      -> Enclave1           : check Registered_Enclave2_Credentials matches Enclave2_Credentials
+  Enclave1      -> TLCC_Enclave1    ++: can_endorse(Enclave2_VK)
+  note right
+    TLCC checks the endorsement policy to determine
+    if the recipient enclave is an eligible endorser.
+  end note
+  note right
+    TLCC should check that the recipient enclave's MRENCLAVE
+    matches the chaincode definition's MRENCLAVE.
+    This could enable key distribution from an old cc version,
+    with a different MRENCLAVE.
+    Notice: in this case there cannot be also a check
+    can_endorse(Enclave1_VK) -- or it should be negated.
+  end note
   return true /'Enclave1 -> TLCC_Enclave1'/
-    note left
-        if false, abort
-    end note
-  Enclave1      -> Enclave1         :   check CC_name, CC_ver, CH_id, org_id in sealed data
-  Enclave1      -> Enclave1         :   check ECC2_S_PK, ECC_E_PK, attestation_report
-  Enclave1      -> Enclave1         :   encrypted_keys_message <- encrypt <SEK, CC_E_SK> with ECC2_E_PK
+  note left
+    if false, abort
+  end note
+  note over Enclave1
+    Regarding Enclave2's chaincode definition, we can only check that CC_Id matches,
+    because the other fields might have changed.
+    Alternatively, we could access the "current" chaincode definition,
+    and check whether Enclave2's matches it.
+  end note
+  Enclave1      -> Enclave1         :   check Enclave2_Credentials.Attested_Data.CC_Parameters.CC_Id against internal one
+  Enclave1      -> Enclave1         :   check Enclave2's keys and Enclave2_VK, Enclave2_EK, attestation_report
```

Comment:
* the enclave has  the ground-truth in the crypto lib
* can_endorse is not well-defined... so maybe?!
* anyway, you can't remove the CC_Id check. Otherwise we would be passing the keys to an enclave registered under a different CC_Id.

Reply:
given that we state the ground truth explicitly for ledeger_enclave, no reader would expect it to be also in the stub_enclave. I'm confused now with my suggestion, though. Seems to be the same as the original text.  I think my suggestion was supposed to be dropping it (i.e., empty suggestion box) and it got somehow filled again with the original text ..